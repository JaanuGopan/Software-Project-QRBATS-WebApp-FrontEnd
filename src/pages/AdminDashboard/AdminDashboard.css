@media screen and (max-width: 1800px){
  .mainHead {
    font-size: x-large;
    margin-top: 5px;
    padding-left: 35px;
  }
  .informationText{
    font-size: large;
  }
  .buttonIcon{
    height: 1.5rem;
    width: 1.5rem;
    size: 1.5rem;
  }
  .NormalButton{
    height: 2.5rem;
    padding: 0.2vw;
    padding-left: 1vw;
  }
  .NormalButtonLabel{
    font-size: larger;
    padding: 1vw;
    margin-bottom: 0px;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 35px;
  }
}

@media screen and (max-width: 768px){
  .mainHead {
    font-size: large;
    margin-top: 10px;
    padding-left: 35px;
  }
  .informationText{
    font-size: large;
  }
  .buttonIcon{
    height: 2rem;
    width: 2rem;
    size: 2rem;

  }
  .NormalButtonLabel{
    font-size: small;
    margin-bottom: 0px;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 35px;
  }
}

@media screen and (max-width: 426px){
  .informationText{
    font-size: medium;
  }
  .NormalButtonLabel{
    font-size: small;
    margin-bottom: 0px;
  }
  .buttonIcon{
    height: 1rem;
    width: 1rem;
    size: 1rem;
  }
  .admin-Dash{
    margin: 0px;
  }
  .SearchEvent{
    margin-right: 0px;
  }
  .mainHead {
    padding-left: 5px;
  }
  .buttonIcon{
    height: 1.5rem;
    width: 1.5rem;
    size: 1.5rem;
  }
}

.admin-Dash{
  flex-grow: 1; 
  height: 90vh;
  width: 100%;
}

.mainHead {
  color: #0D295E;
  font-weight: bold;
  margin-bottom: 0px;
}

.mainInform {
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: wrap;
}

.inform {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: stretch;
  border: 3px solid white;
  border-radius: 10px;
  padding: 2%;
  width: 20vw;
  height: 20vh;
  margin: 1%;
  min-width: 115px;
  background: linear-gradient(#0063A0, rgba(0,100,160,0.8));
}

.inform1 {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.AdminEventList{
  display: flex;
  flex-direction: column;
  margin-top: 10px;
  padding-left: 10px;
  padding-right: 10px;
  padding: 10px;
  width: 100%;
  height: 55vh;
  border-top: 5px solid #0365A1;
  box-shadow: 0.5px 2px 0.5px 0.5px rgba(25, 65, 85, 0.5);
  border-radius: 10px;
  background-color: #FAFAFA;
}

.tableDesign {
 /* position: relative;
  height: 350px;
  display: block;*/
  width: 100%;
  height: -webkit-fill-available;
}

.tableArrangement {
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 45vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.student-tableArrangement{
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 65vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.event-report-tableArrangement{
  display: block;
  overflow: hidden;
  table-layout: auto;
  border-collapse: collapse;
  box-shadow: 0px 5px 5px #ccc;
  border-radius: 10px;
  white-space: nowrap;
  width: 100em;
  max-width: 90%;
  margin: auto;
  overflow-x: auto;
  height: 75vh;
  overflow-y: auto;
  border-bottom: 2px solid #0063A0;
}

.tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.student-tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.event-report-tableArrangement thead{
  position: sticky;
  top: 0;
  background-color: #c1e2f9;
  color: #0D295E;
  border-bottom: 3px solid #0063A0;
  text-align: center;
}

.tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.student-tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.event-report-tableArrangement th{
  padding: 0.8rem;
  text-align: center;
}

.tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.student-tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.event-report-tableArrangement td{
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  padding: 0.5rem;
}

.tableArrangement tbody{
  overflow-y: auto;
}

.student-tableArrangement tbody{
  overflow-y: auto;
}

.event-report-tableArrangement tbody{
  overflow-y: auto;
}

.tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.student-tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.event-report-tableArrangement tbody tr{
  border: 0.01px solid rgba(91, 178, 255, 0.608);
  border-radius: 5px;
}

.tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.student-tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.event-report-tableArrangement tbody tr:hover{
  background-color: azure;
  cursor: pointer;
}

.expand {
  width: 100%;
}

.List-Buttons{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: space-around;
  padding: 1rem;
}

.NormalButton{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  align-items: center;
  justify-content: center;
  color: white;
  cursor: pointer;
  background-color: #0063A0;
  border-radius: 10px;
}

.NormalButton:hover{
  border: 1px solid #025587;
  background-color: #2273a5;
}

.EditButton{
  border-radius: 30px;
  background-color: #0063A0;
  color: white;
  padding: 10px 15px;
}

.EditButton:hover{
  background-color: blue;
}

.Admin-Create-Event-Dashboard{
  position: fixed;
  z-index: 2;
  left: 0;
  top: 0;
  overflow-y: scroll;
  width: 100%;
  height: 100vh;
  padding-top: 5vh;
  padding-bottom: 5vh;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(0, 0, 0, 0.5);
}

.EditIcon{
  padding: 0px;
}

.qr-code{
  background-color: white;
  max-width: 60vw;
}

.SearchEvent{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  padding-right: 10px;
}

.selected-row {
  background-color: rgb(208, 238, 248);
  /* Change the background color to lightblue */
}

.ViewButton{
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  padding: 5px 10px;
  align-items: center;
  justify-content: center;
}

.ViewButtonLabel{
  margin: 0px;
  padding-left: 3px;
}