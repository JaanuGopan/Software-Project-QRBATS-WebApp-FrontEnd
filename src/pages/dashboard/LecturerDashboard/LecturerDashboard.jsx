import { ToggleButton, ToggleButtonGroup } from '@mui/material';
import React, { useContext, useEffect, useState } from 'react';
import { toast } from 'react-toastify';
import EventService from '../../../api/services/EventService';
import LectureService from '../../../api/services/LectureService';
import LocationService from '../../../api/services/LocationService';
import AppContentCard from '../../../components/app-content-card/app-content-card';
import EventTable from '../../../components/tables/event-table';
import LecturesTable from '../../../components/tables/lectures-table';
import WarningPopup from '../../../components/warningPopup/WarningPopup';
import { AuthContext } from '../../../config/AuthProvider';
import AdminUpdateEvent from '../../Event/AdminUpdateEvent';
import LecturesEdit from '../../Lectures/LecturesEdit';
import './LectureDashboard.css';

const LecturerDashboard = () => {
  const [showUpdateLecturePopup, setShowUpdateLecturePopup] = useState(false);
  const [showDeleteLecturePopup, setShowDeleteLecturePopup] = useState(false);
  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);
  const [eventLectureList, setEventLectureList] = useState([]);
  const [selectedLecture, setSelectedLecture] = useState(null);
  const [selectedEventLecture, setSelectedEventLecture] = useState(null);
  const [searchLecture, setSearchLecture] = useState('');
  const [selectTable, setSelectTable] = useState('Lectures');
  const [venuesList, setVenuesList] = useState([]);
  const [showEditLectureWindow, setShowEditLectureWindow] = useState(false);
  const [editLecture, setEditLecture] = useState(null);
  const [processingDeleteLecture, setProcessingDeleteLecture] = useState(false);
  const [lectureList, setLectureList] = useState([]);
  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);

  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const handleReloadEventLectureList = async () => {
    EventService.getEventByUserID(userId)
      .then((list) => {
        setEventLectureList(list);
      })
      .catch((error) => {
        console.error('Error in getting lecture list ', error);
      })
      .finally(() => {});
  };

  const handleReloadLectureList = async () => {
    const response = await LectureService.getAllLecturesByUserId(userId);
    if (response) {
      setLectureList(response);
    }
  };

  const handleEditLecture = (lecture) => {
    if (lecture) {
      setEditLecture(lecture);
      setShowEditLectureWindow(true);
    }
  };

  const handleShowDeleteWindow = (lecture) => {
    setSelectedLecture(lecture);
    setShowDeleteLecturePopup(true);
  };

  const handleDeleteLecture = async () => {
    try {
      setProcessingDeleteLecture(true);
      const response = await LectureService.deleteLecture(selectedLecture.lectureId);
      if (response.status === 200) {
        handleReloadLectureList();
        setSelectedLecture(null);
        toast.success(`Successfully ${selectedLecture.lectureName} Lecture Deleted.`);
        setShowDeleteLecturePopup(false);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Lecture ${selectedLecture.lectureName}.`);
      }
    } finally {
      setProcessingDeleteLecture(false);
    }
  };

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    handleReloadEventLectureList();
    handleGetLocationNameList();
    handleReloadLectureList();
  }, []);

  const handleShowDeleteEventWindow = () => {
    if (selectedEventLecture === null) {
      toast.error('Please Select Event To Delete.');
      return;
    }
    setShowDeleteEventPopup(true);
  };

  const handleDeleteEventLecture = async () => {
    if (selectedEventLecture) {
      try {
        setProcessingDeleteEvent(true);
        const response = await EventService.deleteEvent(selectedEventLecture.eventId);
        if (response.status === 200) {
          handleReloadEventLectureList();
          setSelectedEventLecture(null);
          toast.success(`Successfully ${selectedEventLecture.eventName} Event Deleted.`);
          setShowDeleteEventPopup(false);
        } else if (response.status === 400) {
          toast.error(response.data);
        } else {
          toast.error(`Error In Deleting Lecture ${selectedEventLecture.eventName}.`);
        }
      } finally {
        setProcessingDeleteEvent(false);
      }
    }
  };

  const handleEventLectureClick = (lecture) => {
    setSelectedEventLecture(lecture);
  };

  const handleChange = (e) => {
    setSelectTable(e);
  };

  return (
    <div className="row h-100">
      <div className="col">
        <div>
          <p className="ms-5 mt-2 fw-bold fs-4">{'Lecturer Dashboard'}</p>
        </div>
        <div className="row mx-1">
          <div className="col">
            <div className="row">
              <div className="col-auto d-flex align-items-center pe-0">Select Table :</div>
              <div className="col-auto">
                <ToggleButtonGroup
                  color="primary"
                  value={selectTable}
                  exclusive
                  onChange={(e) => handleChange(e.target.value)}
                  aria-label="Platform"
                >
                  <ToggleButton value="Lectures">Lectures</ToggleButton>
                  <ToggleButton value="Events">Events</ToggleButton>
                </ToggleButtonGroup>
              </div>
            </div>
          </div>
          <div className="col-auto">
            <input
              type="text"
              placeholder="Search..."
              style={{
                width: '150px',
                padding: '3px 40px',
                border: '0.5px solid black',
                borderRadius: '5px',
                textAlign: 'center',
              }}
              onChange={(e) => setSearchLecture(e.target.value)}
            />
          </div>
        </div>
        {selectTable === 'Lectures' ? (
          <AppContentCard hasBottomContents={true}>
            <LecturesTable
              lecturesList={lectureList}
              search={searchLecture}
              onLectureClick={(e) => {
                setSelectedLecture(e);
              }}
              handleLectureUpdate={(lecture) => {
                handleEditLecture(lecture);
              }}
              handleLectureDelete={handleShowDeleteWindow}
            />
          </AppContentCard>
        ) : (
          <AppContentCard hasBottomContents={true}>
            <EventTable
              search={searchLecture}
              handleUpdateEvent={() => setShowUpdateLecturePopup(true)}
              onEventClick={handleEventLectureClick}
              eventList={eventLectureList}
              handleEventDelete={handleShowDeleteEventWindow}
            />
          </AppContentCard>
        )}
        {showUpdateLecturePopup && (
          <div className="Admin-Create-Event-Dashboard">
            <AdminUpdateEvent
              handleCloseEventUpdateWindow={() => setShowUpdateLecturePopup(false)}
              selectedEvent={selectedEventLecture}
              reloadEventList={handleReloadEventLectureList}
              locationNameList={venuesList}
            />
          </div>
        )}
        {showEditLectureWindow && (
          <div className="Admin-Create-Event-Dashboard">
            <LecturesEdit
              selectedLecture={editLecture}
              handleCloseUpdateLectureWindow={() => {
                setShowEditLectureWindow(false);
                setEditLecture(null);
              }}
              handleReload={handleReloadLectureList}
              locationNameList={venuesList}
              reloadLecturesList={handleReloadLectureList}
            />
          </div>
        )}
        {showDeleteLecturePopup && (
          <div className="delete-lecture-container">
            <WarningPopup
              handleCloseWarningWindow={() => setShowDeleteLecturePopup(false)}
              handleOk={handleDeleteLecture}
              titleText={'Are You Sure You Want To Delete This Lecture?'}
              buttonText={'Delete'}
              processing={processingDeleteLecture}
            />
          </div>
        )}
        {showDeleteEventPopup && (
          <div className="delete-lecture-container">
            <WarningPopup
              handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
              handleOk={handleDeleteEventLecture}
              titleText={'Are You Sure You Want To Delete This Event?'}
              buttonText={'Delete'}
              processing={processingDeleteEvent}
            />
          </div>
        )}
      </div>
    </div>
  );
};

export default LecturerDashboard;
