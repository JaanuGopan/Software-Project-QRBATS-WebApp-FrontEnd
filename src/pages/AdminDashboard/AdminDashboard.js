import React, { useState, useEffect } from "react";
import "./AdminDashboard.css";
import Table from "../../components/layout/AdminDashboardComponent/Table";
import TotalCount from "../../components/layout/AdminDashboardComponent/TotalCount";
import { FaUsers } from "react-icons/fa";
import { PiUsersFourFill } from "react-icons/pi";
import { FaSchool } from "react-icons/fa6";
import { IoNewspaperSharp } from "react-icons/io5";
import NormalButton from "../../components/layout/AdminDashboardComponent/NormalButton";
import { MdCreateNewFolder } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import AdminUpdateEvent from "../Event/AdminUpdateEvent";
import AdminEventCreation from "../Event/AdminEventCreation";
import DeleteEventService from "../../api/services/DeleteEventService";
import FetchEventsService from "../../api/services/FetchEventsService";

const AdminDashboard = () => {
  const [eventCreatePopUpWindow, setEventCreatePopUpWindow] = useState(false);
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState("");

  useEffect(() => {
    // Fetch the list of events from the API using the new class
    handleReloadEventList();
  }, []);

  // Function to handle event click
  const handleEventClick = (event) => {
    setSelectedEvent(event);
    // Do whatever you want with the selected event data
    console.log("Selected Event:", event);
  };

  const handleDelete = async () => {
    try {
      const response = await DeleteEventService.deleteEvent(
        selectedEvent.eventId
      );
      // After deleting, you may want to update the event list
      // Fetch the updated event list
      handleReloadEventList();
    } catch (error) {
      console.log("error " + error);
    }
  };

  const handleReloadEventList = async () => {
    FetchEventsService.fetchEvents()
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error("Error fetching events:", error);
      });
  };

  return (
    <div className="admin-Dash">
      <p className="mainHead">Admin Dashboard</p>
      <div className="mainInform">
        <TotalCount
          total={"20"}
          countIcon={
            <FaUsers
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Staffs"}
        />
        <TotalCount
          total={"201"}
          countIcon={
            <PiUsersFourFill
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Students"}
        />
        <TotalCount
          total={"08"}
          countIcon={
            <FaSchool
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Departments"}
        />
        <TotalCount
          total={"350"}
          countIcon={
            <IoNewspaperSharp
              style={{ color: "white", padding: "2%", fontSize: "250%" }}
            />
          }
          countTitle={"Total Modules"}
        />
      </div>
      <div className="SearchEvent">
        <p className="mainHead">Upcoming Events</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: "150px",
            padding: "3px 40px",
            border: "0.5px solid black",
            borderRadius: "5px",
            textAlign: "center",
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="AdminEventList">
        <Table
          search={search}
          handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
          onEventClick={handleEventClick}
          eventList={eventList} // Pass the eventList prop here
        />

        <div className="List-Buttons">
          <NormalButton
            handleClick={() => setEventCreatePopUpWindow(true)}
            title={"Create"}
            titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
          />
          <NormalButton
            title={"Delete"}
            handleClick={handleDelete}
            titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
          />
        </div>
      </div>
      {eventCreatePopUpWindow && (
        <div
          handleClick={() => setEventCreatePopUpWindow(false)}
          className="Admin-Create-Event-Dashboard"
        >
          <AdminEventCreation
            handlecloseCreateEventWindow={() =>
              setEventCreatePopUpWindow(false)
            }
            reloadEventList={handleReloadEventList}
          />
        </div>
      )}
      {eventUpdatePopUpWindow && (
        <div
          handleClick={() => setEventCreatePopUpWindow(false)}
          className="Admin-Create-Event-Dashboard"
        >
          <AdminUpdateEvent
            handlecloseCreateEventWindow={() =>
              setEventUpdatePopUpWindow(false)
            }
            selectedEvent={selectedEvent}
            reloadEventList={handleReloadEventList}
          />
        </div>
      )}
    </div>
  );
};

export default AdminDashboard;
