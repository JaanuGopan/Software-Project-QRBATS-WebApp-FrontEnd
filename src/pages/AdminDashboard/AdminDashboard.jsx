import React, { useState, useEffect, useContext } from 'react';
import './AdminDashboard.css';
import Table from '../../components/layout/AdminDashboardComponent/Table';
import NormalButton from '../../components/layout/AdminDashboardComponent/NormalButton';
import { MdCreateNewFolder } from 'react-icons/md';
import { RiDeleteBin5Fill } from 'react-icons/ri';
import AdminUpdateEvent from '../Event/AdminUpdateEvent';
import AdminEventCreation from '../Event/AdminEventCreation';
import LocationService from '../../api/services/LocationService';
import EventService from '../../api/services/EventService';
import { toast, ToastContainer } from 'react-toastify';
import WarningPopup from '../../components/warningPopup/WarningPopup';
import { AuthContext } from '../../config/AuthProvider';

const AdminDashboard = () => {
  const [eventCreatePopUpWindow, setEventCreatePopUpWindow] = useState(false);
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState('');
  const [venuesList, setVenuesList] = useState([]);

  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    // Fetch the list of events from the API using the new class
    handleReloadEventList();
    handleGetLocationNameList();
  }, []);

  // Function to handle event click
  const handleEventClick = (event) => {
    setSelectedEvent(event);
    // Do whatever you want with the selected event data
    console.log('Selected Event:', event);
  };

  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);

  const showDeleteEventWarning = () => {
    if (selectedEvent === null) {
      toast.error('Please Select Event To Delete.');
      return;
    }
    setShowDeleteEventPopup(true);
  };

  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);

  const handleDelete = async () => {
    try {
      setProcessingDeleteEvent(true);
      const response = await EventService.deleteEvent(selectedEvent.eventId);
      if (response.status === 200) {
        handleReloadEventList();
        setSelectedEvent(null);
        toast.success(`Successfully ${selectedEvent.eventName} Event Deleted.`);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Event ${selectedEvent.eventName}.`);
      }
    } catch (error) {
      console.log('error ' + error);
    } finally {
      setProcessingDeleteEvent(false);
      setShowDeleteEventPopup(false);
    }
  };

  const handleReloadEventList = async () => {
    EventService.getEventByUserID(userId)
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error('Error fetching events:', error);
      });
  };

  return (
    <div className="admin-Dash">
      <ToastContainer />
      <p className="mainHead">Admin Dashboard</p>
      <div className="mainInform"></div>
      <div className="SearchEvent">
        <p className="mainHead">Upcoming Events</p>
        <input
          type="text"
          placeholder="Search..."
          style={{
            width: '150px',
            padding: '3px 40px',
            border: '0.5px solid black',
            borderRadius: '5px',
            textAlign: 'center',
          }}
          onChange={(e) => setSearch(e.target.value)}
        />
      </div>
      <div className="AdminEventList">
        <Table
          search={search}
          handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
          onEventClick={handleEventClick}
          eventList={eventList} // Pass the eventList prop here
        />

        <div className="List-Buttons">
          <NormalButton
            handleClick={() => setEventCreatePopUpWindow(true)}
            title={'Create'}
            titlewithiconicon={<MdCreateNewFolder className="buttonIcon" />}
            id={'Admincreate'}
          />
          <NormalButton
            title={'Delete'}
            handleClick={showDeleteEventWarning}
            titlewithiconicon={<RiDeleteBin5Fill className="buttonIcon" />}
          />
        </div>
      </div>
      {eventCreatePopUpWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <AdminEventCreation
            handleCloseCreateEventWindow={() => setEventCreatePopUpWindow(false)}
            reloadEventList={handleReloadEventList}
            locationList={venuesList}
            showCloseButton={true}
          />
        </div>
      )}
      {eventUpdatePopUpWindow && (
        <div className="Admin-Create-Event-Dashboard">
          <AdminUpdateEvent
            handleCloseEventUpdateWindow={() => setEventUpdatePopUpWindow(false)}
            selectedEvent={selectedEvent}
            reloadEventList={handleReloadEventList}
            locationNameList={venuesList}
          />
        </div>
      )}

      {showDeleteEventPopup && (
        <div className="delete-event-container">
          <WarningPopup
            handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
            handleOk={handleDelete}
            titleText={'Are You Sure You Want To Delete This Event?'}
            buttonText={'Delete'}
            processing={processingDeleteEvent}
          />
        </div>
      )}
    </div>
  );
};

export default AdminDashboard;
