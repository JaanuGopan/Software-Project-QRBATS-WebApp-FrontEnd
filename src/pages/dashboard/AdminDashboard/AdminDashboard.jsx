import React, { useContext, useEffect, useState } from 'react';
import { FaPlus } from 'react-icons/fa';
import { toast } from 'react-toastify';
import EventService from '../../../api/services/EventService';
import LocationService from '../../../api/services/LocationService';
import AppContentCard from '../../../components/app-content-card/app-content-card';
import NormalButton from '../../../components/buttons/NormalButton';
import EventTable from '../../../components/tables/event-table';
import WarningPopup from '../../../components/warningPopup/WarningPopup';
import { AuthContext } from '../../../config/AuthProvider';
import AdminEventCreation from '../../Event/AdminEventCreation';
import AdminUpdateEvent from '../../Event/AdminUpdateEvent';
import './AdminDashboard.css';

const AdminDashboard = () => {
  const [eventUpdatePopUpWindow, setEventUpdatePopUpWindow] = useState(false);
  const [selectedEvent, setSelectedEvent] = useState(null);
  const [eventList, setEventList] = useState([]);
  const [search, setSearch] = useState('');
  const [venuesList, setVenuesList] = useState([]);
  const [showDeleteEventPopup, setShowDeleteEventPopup] = useState(false);
  const [processingDeleteEvent, setProcessingDeleteEvent] = useState(false);
  const [showEventCreationModal, setShowEventCreationModal] = useState(false);

  const { user } = useContext(AuthContext);
  const { userId } = user || {};

  const handleGetLocationNameList = async () => {
    const response = await LocationService.getAllLocationNames();
    setVenuesList(response);
  };

  useEffect(() => {
    handleReloadEventList();
    handleGetLocationNameList();
  }, []);

  const handleEventClick = (event) => {
    setSelectedEvent(event);
  };

  const showDeleteEventWarning = () => {
    setShowDeleteEventPopup(true);
  };

  const handleDelete = async () => {
    try {
      setProcessingDeleteEvent(true);
      const response = await EventService.deleteEvent(selectedEvent.eventId);
      if (response.status === 200) {
        handleReloadEventList();
        setSelectedEvent(null);
        toast.success(`Successfully ${selectedEvent.eventName} Event Deleted.`);
      } else if (response.status === 400) {
        toast.error(response.data);
      } else {
        toast.error(`Error In Deleting Event ${selectedEvent.eventName}.`);
      }
    } catch (error) {
      console.error('error ' + error);
    } finally {
      setProcessingDeleteEvent(false);
      setShowDeleteEventPopup(false);
    }
  };

  const handleReloadEventList = async () => {
    EventService.getEventByUserID(userId)
      .then((events) => {
        setEventList(events);
      })
      .catch((error) => {
        console.error('Error fetching events:', error);
      });
  };

  return (
    <div className="row">
      <div className="col py-2">
        <div className="row my-1 align-items-center justify-content-between">
          <div className="col-auto ms-4 fs-3 fw-bold ">Upcoming Events</div>
          <div className="col-auto">
            <div className="row align-items-center">
              <div className="col-auto">
                <NormalButton
                  title={'Create Event'}
                  handleClick={() => setShowEventCreationModal(true)}
                  icon={<FaPlus className="buttonIcon" />}
                />
              </div>
              <div className="col-auto">
                <input
                  type="text"
                  placeholder="Search..."
                  style={{
                    width: '150px',
                    padding: '3px 40px',
                    border: '0.5px solid black',
                    borderRadius: '5px',
                    textAlign: 'center',
                  }}
                  onChange={(e) => setSearch(e.target.value)}
                />
              </div>
            </div>
          </div>
        </div>
        <AppContentCard>
          <EventTable
            search={search}
            handleUpdateEvent={() => setEventUpdatePopUpWindow(true)}
            onEventClick={handleEventClick}
            eventList={eventList}
            handleEventDelete={showDeleteEventWarning}
          />
        </AppContentCard>
        {showEventCreationModal && (
          <div className="Admin-Create-Event-Dashboard">
            <AdminEventCreation
              handleCloseCreateEventWindow={() => setShowEventCreationModal(false)}
              reloadEventList={handleReloadEventList}
              locationList={venuesList}
              showCloseButton={true}
            />
          </div>
        )}
        {eventUpdatePopUpWindow && (
          <div className="Admin-Create-Event-Dashboard">
            <AdminUpdateEvent
              handleCloseEventUpdateWindow={() => setEventUpdatePopUpWindow(false)}
              selectedEvent={selectedEvent}
              reloadEventList={handleReloadEventList}
              locationNameList={venuesList}
            />
          </div>
        )}
        {showDeleteEventPopup && (
          <div className="delete-event-container">
            <WarningPopup
              handleCloseWarningWindow={() => setShowDeleteEventPopup(false)}
              handleOk={handleDelete}
              titleText={'Are You Sure You Want To Delete This Event?'}
              buttonText={'Delete'}
              processing={processingDeleteEvent}
            />
          </div>
        )}
      </div>
    </div>
  );
};

export default AdminDashboard;
