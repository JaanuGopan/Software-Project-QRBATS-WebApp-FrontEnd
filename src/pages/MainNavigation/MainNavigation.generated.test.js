import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";


import AdminDashboard from "../AdminDashboard/AdminDashboard";
import { PiListDashesFill } from "react-icons/pi";
import StaffNavBar from "../../components/layout/StaffDashboardComponents/StaffNavBar";
import AdminSideBar from "../../components/layout/AdminDashboardComponent/AdminSideBar";
import EventCreateDashboard from "../Event/EventCreateDashboard";
import StudentDashboard from "../Student/StudentDashboard";
import Setting from "../Setting/Setting";
import StaffDashboard from "../Staff/StaffDashboard";
import EventReport from "../Event/EventReport";
import { useDispatch, useSelector } from "react-redux";
import { login, selectUser } from "../../redux/features/userSlice";
import { useNavigate } from "react-router-dom";
import LecturerSideBar from "../../components/layout/AdminDashboardComponent/LecturerSideBar";
import LecturerDashboard from "../LecturerDashboard/LecturerDashboard";
import ModulePage from "../Module/ModulePage";
import MainNavigationPage from "./MainNavigation";

jest.mock("./MainNavigation.css");
jest.mock("../StaffMainNavigation/StaffMainNavigation.css");
jest.mock("../AdminDashboard/AdminDashboard");
jest.mock("react-icons/pi");
jest.mock("../../components/layout/StaffDashboardComponents/StaffNavBar");
jest.mock("../../components/layout/AdminDashboardComponent/AdminSideBar");
jest.mock("../Event/EventCreateDashboard");
jest.mock("../Student/StudentDashboard");
jest.mock("../Setting/Setting");
jest.mock("../Staff/StaffDashboard");
jest.mock("../Event/EventReport");
jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("react-router-dom");
jest.mock("../../components/layout/AdminDashboardComponent/LecturerSideBar");
jest.mock("../LecturerDashboard/LecturerDashboard");
jest.mock("../Module/ModulePage");

const renderTree = tree => renderer.create(tree);
describe('<MainNavigationPage>', () => {
  it('should render component', () => {
    expect(renderTree(<MainNavigationPage 
    />).toJSON()).toMatchSnapshot();
  });
  
});