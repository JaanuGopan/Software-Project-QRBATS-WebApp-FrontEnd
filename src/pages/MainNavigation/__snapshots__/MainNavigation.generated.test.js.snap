// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainNavigationPage> should render component 1`] = `
<div
  className="staff-Main"
>
  <div
    className="menuButton"
    onClick={[Function]}
  />
  <div
    className="staff-Submain"
  />
</div>
`;
